UniversalAlgebra/Structures/Graphs.lagda

```diff
@@ -52,34 +53,35 @@ Gr-sig 𝐹 𝑅 = record { symbol = symbol 𝑅 ⊎ symbol 𝐹
  ar (inr 𝑓) = (arity 𝐹) 𝑓 ⊎ 𝟙
 
 
-module _ {𝐹 𝑅 : signature} where
+module _ {𝐹 𝑅 : signature}{α ρ : Level} where
 
- Gr : structure 𝐹 {ℓ₀} 𝑅 {ℓ₀} → structure Sig∅ {ℓ₀} (Gr-sig 𝐹 𝑅) {ℓ₀}
+ Gr : structure 𝐹 {α} 𝑅 {ρ} → structure Sig∅ {α} (Gr-sig 𝐹 𝑅) {α ⊔ ρ}
  Gr 𝑨 = record { carrier = carrier 𝑨 ; op = λ () ; rel = split }
   where
-  split : (s : symbol 𝑅 ⊎ symbol 𝐹) → Rel (carrier 𝑨) (arity (Gr-sig 𝐹 𝑅) s) {ℓ₀}
-  split (inl 𝑟) arg = rel 𝑨 𝑟 arg
-  split (inr 𝑓) args = op 𝑨 𝑓 (args ∘ inl) ≡ args (inr 𝟙.𝟎)
+  split : (s : symbol 𝑅 ⊎ symbol 𝐹) → Rel (carrier 𝑨) (arity (Gr-sig 𝐹 𝑅) s) {α ⊔ ρ}
```

Comment:
Aha! To me, this is a symptom that the "usual" definition is broken, i.e. it makes implicit size assumptions that appear later when you try to use it. It tries to 'squish' two different kinds of relations into a single spot. All the troubles would disappear if the graph had 2 sorts of edges. [Another option is to associate 2 graphs to a structure, but that seems worse.]  In set theory, there is no issue here at all, but that's because set theory doesn't care about either types or size.

Reply:
Very interesting.  I'm seriously grateful that you are looking at all the code and explaining what's really going on.  Your comments/advice are invaluable!  Thank you so much Jacques!!